config/routes.rb

```diff
@@ -0,0 +1,30 @@
+Rails.application.routes.draw do
+  root 'users#index'
+
+  resources :markets, except: [:destroy] do
+    resources :vendors, except: [:index] do
+      resources :sales, only: [:create, :new]
+      resources :products, except: [:show]
+    end
+  end
+
+  get  "/all_markets"        => "users#show"
+  get  "/all_markets/:id"    => "users#market_details"
```

Comment:
This should be using markets/:id, which is served by the resources :markets route above and thus should be removed.